lib/icinga/apievents.cpp

```diff
@@ -391,3 +393,40 @@ void ApiEvents::DowntimeTriggeredHandler(const Downtime::Ptr& downtime)
 
 	inboxes.Push(std::move(result));
 }
+
+void ApiEvents::OnActiveChangedHandler(const ConfigObject::Ptr& object, const Value&)
+{
+	if (object->IsActive()) {
+		ApiEvents::GetProcessResultHelper(object, EventType::ObjectCreated, "ObjectCreated");
+	} else if (!object->IsActive() && !object->GetExtension("ConfigObjectDeleted").IsEmpty()) {
+		ApiEvents::GetProcessResultHelper(object, EventType::ObjectDeleted, "ObjectDeleted");
+	}
+}
+
+void ApiEvents::OnVersionChangedHandler(const ConfigObject::Ptr& object, const Value&)
+{
+	ApiEvents::GetProcessResultHelper(object, EventType::ObjectModified, "ObjectModified");
+}
+
+void ApiEvents::GetProcessResultHelper(const ConfigObject::Ptr &object, const EventType &eventType, const String& eventQueue) {
```

Comment:
``ProcessResult`` Because the results of the new events are processed here and ``Helper`` because the function is not a real event handler but only a helper function. If you want a different name then tell me and I will change it.

Reply:
I can't really follow the reasoning here. I'm especially irritated by the "Get" prefix. I'd suggest something like `SendObjectChangeEvent(Helper)`.